docs/sources/panels/queries.md

```diff
@@ -0,0 +1,119 @@
+
+Because of the difference between query languages, data sources may have query editors that look different. Here are two examples of query editors:
+
+**InfluxDB query editor**
+
+{{< docs-imagebox img="/img/docs/queries/influxdb-query-editor-7-0.png" class="docs-image--no-shadow" max-width="1000px" >}}
+
+**Prometheus (PromQL) query editor**
+
+{{< docs-imagebox img="/img/docs/queries/prometheus-query-editor-7-0.png" class="docs-image--no-shadow" max-width="1000px" >}}
+
+## Query syntax
+
+Data sources have different query languages and syntaxes to ask for the data. Here are two query examples:
+
+**PostgreSQL**
+
+```
+SELECT hostname FROM host  WHERE region IN($region)
+```
+
+**PromQL**
+
+```
+query_result(max_over_time(<metric>[${__range_s}s]) != <state>)
+```
+
+For more information about writing a query for your data source, refer to the specific [Grafana data source]({{< relref "../features/datasources/_index.md" >}}) documentation.
+
+## Query tab UI
+
+The Query tab consists of the following elements:
+
+* Data source selector
+* Data source options
+* Query inspector button
+* Query editor list
+
+{{< docs-imagebox img="/img/docs/queries/query-editor-7-0.png" class="docs-image--no-shadow" max-width="1000px" >}}
+
+### Data source selector
+
+The data source selector is a drop-down list. Click it to select a data source you have added. When you create a panel, Grafana automatically selects your default data source. For more information about adding data sources, refer to [Add a data source]({{< relref "../features/datasources/add-a-data-source.md" >}}).
+
+{{< docs-imagebox img="/img/docs/queries/data-source-selector-7-0.png" class="docs-image--no-shadow" max-width="250px" >}}
+
+In addition to the data sources that you have configured in your Grafana, there are three special data sources available:
+
+* **Grafana -** A built-in data source that generates random walk data. Useful for testing visualizations and running experiments.
+* **Mixed -** Select this to query multiple data sources in the same panel. When this data source is selected, Grafana allows you to select a data source for every new query that you add.
+  * The first query will use the data source that was selected before you selected **Mixed**.
+  * You cannot change an existing query to use the Mixed Data Source.
+* **Dashboard -** Select this to use a result set from another panel in the same dashboard.
+
+### Data source options
+
+Click **Options** next to the data source selector to see settings for your selected data source. Changes you make here affect only this panel.
+
+{{< docs-imagebox img="/img/docs/queries/data-source-options-7-0.png" class="docs-image--no-shadow" max-width="400px" >}}
+
+Grafana sets defaults that are shown in dark gray text. Changes are displayed in white text. To return a field to the default setting, delete the white text from the field.
+
+Panel data source options:
+
```

Comment:
@torkelo or @dprokop , please enter definitions for these fields.

Reply:
@daniellee , @torkelo ?

Reply:
### max data points

If the data source supports it, sets the maximum numbers of data points for each series returned. If the query returns more data points than the `max data points` setting then the data source will consolidate them (reduce the number of points returned by aggregating them together by average or max or other function). There are two main reasons for limiting the number of points, performance and smoothing the line. The default value is the width (or number of pixels) of the graph as there is no point in having more data points than the graph panel can display.

With streaming data, the max data points value is used for the rolling buffer. (Streaming is a continuous flow of data and buffering is a way of dividing the stream into chunks). Loki streams data in the live tailing mode.

This article describes how they can be confusing - the default consolidation function for Graphite is the average when using maxdatapoints and this can be surprising if you are expecting something else: https://addshore.com/2018/09/grafana-graphite-and-maxdatapoints-confusion-for-totals/

Reply:
### interval

The interval is a time span can be used when aggregating or grouping by time. Grafana automatically calculates an appropriate interval and it can be used as a variable in templated queries. The variable is either in seconds: `$__interval` or in milliseconds: `$__interval_ms`. It is typically used in aggregation functions like sum or average. For example, a Prometheus query using the interval variable: `rate(http_requests_total[$__interval])`.

This automatic interval is calculated based on the width of the graph. If the user zooms out a lot then the interval becomes greater, resulting in a more coarse grained aggregation whereas if the use zooms in then the interval decreases resulting in a more fine grained aggregation.

More details here: https://github.com/grafana/grafana/blob/add-queries.md/docs/sources/variables/global-variables.md

Reply:
### Min interval

Sets a minimum limit for the automatically calculated interval. Typically set to the minimum scrape interval (using Prometheus terminology here). If a data point is saved every 15 seconds then there is no point in having an interval that is lower than that. Another use case is set it to a higher minimum than the scrape interval to get more coarse grained, performant queries.

Reply:
### Relative time

You can override the relative time range for individual panels, causing them to be different than what is selected in the Dashboard time picker in the upper right. This allows you to show metrics from different time periods or days on the same dashboard.

Reply:
### Time shift

The time shift function is another way to override the time range for individual panels. It only works with relative time ranges and allows you to adjust the time range. For example, shifting the time range for the panel to be 2 hours earlier than the dashboard time picker.

https://grafana.com/docs/grafana/latest/reference/timerange/#panel-time-overrides-and-timeshift

Reply:
### Cache timeout

If your time series store has a query cache, this option can override the default cache timeout. Specified as a numeric value in seconds. 

Here is the original issue. The use case was to override the Graphite cache time of 1m to be able to more up to date data.

https://github.com/grafana/grafana/issues/266